src/components/shared/MarkerTooltipContents.js

```diff
@@ -34,6 +47,50 @@ function getMarkerDetails(data: MarkerPayload): React$Element<*> | null {
           </div>
         );
       }
+      case 'GCMinor': {
+        if (data.nursery && data.nursery.reason) {
```

Comment:
These marker payloads are not covered by Flow types. We try to type everything that we get from the Gecko Profiler so that way it's really easy to update our code when any profile formats change. Flow can infer the correct types based on these switches, and it will fail when we make incompatible changes to the payload formats. Here's my suggested approach for adding these definitions. Please let me know if you need any additional direction on using Flow.

```diff
diff --git a/src/components/shared/MarkerTooltipContents.js b/src/components/shared/MarkerTooltipContents.js
index 1637983..3bcc90c 100644
--- a/src/components/shared/MarkerTooltipContents.js
+++ b/src/components/shared/MarkerTooltipContents.js
@@ -11,7 +11,7 @@ import formatTimeLength from '../../utils/format-time-length';
 import type { TracingMarker } from '../../types/profile-derived';
 import type { MarkerPayload } from '../../types/markers';
 
-function markerDetail(key, label, value) {
+function markerDetail(key: string, label: string, value: string) {
   if (value) {
     return [
       <div className="tooltipLabel" key="{key}">
diff --git a/src/types/markers.js b/src/types/markers.js
index b756e19..2a2030c 100644
--- a/src/types/markers.js
+++ b/src/types/markers.js
@@ -70,6 +70,28 @@ export type DOMEventMarkerPayload = {
   phase: number,
 };
 
+export type GCMinorMarkerPayload = {
+  type: 'GCMinor',
+  startTime: Milliseconds,
+  endTime: Milliseconds,
+  nursery: {|
+    reason: string,
+  |},
+};
+
+export type GCMajorMarkerPayload = {
+  type: 'GCMajor',
+  startTime: Milliseconds,
+  endTime: Milliseconds,
+  timings: {|
+    // TODO
+  |},
+};
+
+export type GCSliceMarkerPayload = {
+  // TODO
+};
+
 /**
  * The union of all the different marker payloads that perf.html knows about, this is
  * not guaranteed to be all the payloads that we actually get from the profiler.
@@ -79,4 +101,7 @@ export type MarkerPayload =
   | UserTimingMarkerPayload
   | PaintProfilerMarkerTracing
   | DOMEventMarkerPayload
+  | GCMinorMarkerPayload
+  | GCMajorMarkerPayload
+  | GCSliceMarkerPayload
   | null;
```